tsconfig.json

```diff
@@ -2,17 +2,19 @@
   "extends": "aegir/src/config/tsconfig.aegir.json",
   "compilerOptions": {
     "jsx": "react",
-    "types": ["node"],
+    "types": ["node", "mocha"],
     "outDir": "dist",
     "target": "ES6",
     "allowSyntheticDefaultImports": true,
     "moduleResolution": "NodeNext",
+    "module": "es2022",
     "typeRoots": [
       "src/types"
     ],
   },
   "include": [
     "src",
+    "test",
```

Comment:
I'm more than fine with it. it's done in thousands of packages, but we can also change package.json to only pack up "dist/src"

Also, eslint rules & typechecking should apply to tests, otherwise you end up with ugly code testing your clean code.

Reply:
https://github.com/ipfs-shipyard/ignite-metrics/pull/64/commits/ef9d4cc28add878ad50bac2a6f636b1810a531e2